src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1047 @@
+            return out;
+        } catch (IOException ioe) {
+            throw new UncheckedIOException(ioe.getMessage(), ioe);
+        }
+    }
+
+    /**
+     * Returns the four hexadecimal characters for the {@code char} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a {@code char} value
+     * @return the four hexadecimal characters for the {@code char} value
+     */
+    public String toHexDigits(char value) {
+        return toHexDigits((short)value);
+    }
+
+    /**
+     * Returns the four hexadecimal characters for the {@code short} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a {@code short} value
+     * @return the four hexadecimal characters for the {@code short} value
+     */
+    public String toHexDigits(short value) {
+        byte[] rep = new byte[4];
+        rep[0] = (byte)toHighHexDigit((byte)(value >> 8));
+        rep[1] = (byte)toLowHexDigit((byte)(value >> 8));
+        rep[2] = (byte)toHighHexDigit((byte)value);
+        rep[3] = (byte)toLowHexDigit((byte)value);
+
+        try {
+            return jla.newStringNoRepl(rep, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    /**
+     * Returns the eight hexadecimal characters for the {@code int} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value an {@code int} value
+     * @return the eight hexadecimal characters for the {@code int} value
+     */
+    public String toHexDigits(int value) {
+        byte[] rep = new byte[8];
+        rep[0] = (byte)toHighHexDigit((byte)(value >> 24));
+        rep[1] = (byte)toLowHexDigit((byte)(value >> 24));
+        rep[2] = (byte)toHighHexDigit((byte)(value >> 16));
+        rep[3] = (byte)toLowHexDigit((byte)(value >> 16));
+        rep[4] = (byte)toHighHexDigit((byte)(value >> 8));
+        rep[5] = (byte)toLowHexDigit((byte)(value >> 8));
+        rep[6] = (byte)toHighHexDigit((byte)value);
+        rep[7] = (byte)toLowHexDigit((byte)value);
+
+        try {
+            return jla.newStringNoRepl(rep, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    /**
+     * Returns the sixteen hexadecimal characters for the {@code long} value
+     * considering it to be unsigned.
```

Comment:
What does this means? And why is it necessary for long when it wasn't necessary for other signed types (byte, short, int). I'd vote for removing `considering it to be unsigned` there - or else add it everywhere else, but then you would have to explain what it means and what you would print if you considered it signed.

Reply:
Will remove.  All java primitive types are signed.  When formatted, there is no representation of the sign.